command-line/lib/Command/Address/Reward.hs

```diff
@@ -70,24 +65,43 @@ mod liftCmd = command "stake" $
             ])
   where
     parser = Cmd
-        <$> credentialOpt
-        <*> networkTagOpt Shelley
+        <$> networkTagOpt Shelley
 
 run :: Cmd -> IO ()
-run Cmd{networkTag,credentialType} = do
-    case (mkNetworkDiscriminant . fromIntegral . unNetworkTag) networkTag of
-        Left e -> die (fmt $ build e)
-        Right discriminant -> do
-            addr <- case credentialType of
-                CredentialFromKey -> do
-                    xpub <- hGetXPub stdin
-                    let credential = DelegationFromKey $ Shelley.liftXPub xpub
+run Cmd{networkTag} = do
+    discriminant <- fromNetworkTag networkTag
+    (hrp, bytes) <- hGetBech32 stdin allowedPrefixes
+    addr <- stakeAddressFromBytes discriminant bytes hrp
+    hPutBytes stdout (unAddress addr) (EBech32 stakeHrp)
+  where
+    stakeHrp
+        | networkTag == shelleyTestnet = CIP5.stake_test
+        | otherwise = CIP5.stake
+
+    -- TODO: Also allow `XXX_vk` prefixes. We don't need the chain code to
+    -- construct a payment credential. This will however need some additional
+    -- abstraction over `xpubFromBytes` but I've done enough yake-shaving at
+    -- this stage, so leaving this as an item for later.
+    allowedPrefixes =
+        [ CIP5.addr_xvk
+        , CIP5.stake_xvk
+        , CIP5.script_xvk
```

Comment:
the same here... of course we can accept this and if detected then we need to hash it like provided in hashing before putting to DelegationFromScript